src/OpenFeature.Contrib.Providers.Flagd/JsonEvaluator.cs

```diff
@@ -0,0 +1,223 @@
+
+        private string _selector;
+
+        private readonly JsonLogicEvaluator _evaluator = new JsonLogicEvaluator(EvaluateOperators.Default);
+
+        internal JsonEvaluator(string selector)
+        {
+            _selector = selector;
+        }
+
+        internal void Sync(FlagConfigurationUpdateType updateType, string flagConfigurations)
+        {
+            var flagConfigsMap = JsonConvert.DeserializeObject<FlagSyncData>(flagConfigurations);
+
+            switch (updateType)
+            {
+                case FlagConfigurationUpdateType.ALL:
+                    _flags = flagConfigsMap.Flags;
+                    break;
+                case FlagConfigurationUpdateType.ADD:
+                    foreach (var keyAndValue in flagConfigsMap.Flags)
+                    {
+                        _flags[keyAndValue.Key] = keyAndValue.Value;
+                    }
+                    break;
+                case FlagConfigurationUpdateType.UPDATE:
+                    foreach (var keyAndValue in flagConfigsMap.Flags)
+                    {
+                        _flags[keyAndValue.Key] = keyAndValue.Value;
+                    }
+                    break;
+                case FlagConfigurationUpdateType.DELETE:
+                    foreach (var keyAndValue in flagConfigsMap.Flags)
+                    {
+                        _flags.Remove(keyAndValue.Key);
+                    }
+                    break;
+
+            }
+        }
+
+        public ResolutionDetails<bool> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+        {
+            return ResolveValue(flagKey, defaultValue, context);
+        }
+
+        public ResolutionDetails<string> ResolveStringValue(string flagKey, string defaultValue, EvaluationContext context = null)
+        {
+            return ResolveValue(flagKey, defaultValue, context);
+        }
+
+        public ResolutionDetails<int> ResolveIntegerValue(string flagKey, int defaultValue, EvaluationContext context = null)
+        {
+            return ResolveValue(flagKey, defaultValue, context);
+        }
+
+        public ResolutionDetails<double> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+        {
+            return ResolveValue(flagKey, defaultValue, context);
+        }
+
+        public ResolutionDetails<Value> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+        {
+            return ResolveValue(flagKey, defaultValue, context);
+        }
+
+        private ResolutionDetails<T> ResolveValue<T>(string flagKey, T defaultValue, EvaluationContext context = null)
+        {
+            // check if we find the flag key
+            var reason = Reason.Default;
+            if (_flags.TryGetValue(flagKey, out var flagConfiguration))
+            {
+                reason = Reason.Static;
```

Comment:
We have to check for `DISABLED` state of the flag. Maybe you can refer Java impl [1] or go impl [2] to see if all checks map here :) 

[1] - https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java#L158

[2] - https://github.com/open-feature/flagd/blob/main/core/pkg/evaluator/json.go#L292